backend/src/main/java/de/amos/apachepulsarui/service/TenantService.java

```diff
@@ -29,20 +29,45 @@ public List<String> getAllNames() throws PulsarApiException {
         }
     }
 
+    @Cacheable("tenats.getAll")
+    public List<TenantDto> getAllFiltered(List<String> tenants) throws PulsarApiException {
+        try {
+            List<String> tenantNames = pulsarAdmin.tenants().getTenants();
+
+            if (tenants == null || tenants.isEmpty()) {
+                return tenantNames.stream()
+                        .map(name -> TenantDto.create(getTenantInfo(name), name))
+                        .toList();
+            }
+
+            List<String> filteredNames = tenantNames.stream()
+                    .filter(tenant -> tenants.stream().anyMatch(t -> t.equals(tenant)))
```

Comment:
could use `contains`, but only for readability, doesn't change the functionality